WordPress/gutenberg
# Button: Improve the aria-disabled focus style

Fixes https://github.com/WordPress/gutenberg/issues/62278

## What?

The focus style for `aria-disabled` buttons is inconsistent across the button variants. Also, for some variants, it doesn't meet the WCAG contrast guidelines. For screenshots of the inconsistencies and insufficient color contrast of the focus style, please see the issue https://github.com/WordPress/gutenberg/issues/62278.


## Why?

For better accessibility, while the button itself may have some styling to visually indicate it's 'disabled', the focus style needs to be fully visible. The focus outline is a positional marker to allow a user to identify their current position in the page, and that needs to meet contrast guidelines.

## How?

- Removes any CSS `opacity` property for the button element styling. The only opacity value is now used for the SVG icon, when disabled.
- The disabled / aria-disabled color is now always `$gray-600: #949494` except for the `is-destructive` button.

### There are two visible changes:
- Default button, when disabled: it now uses `$gray-600 #949494` instead of opacity so it's slightly darker but it's consistent with the gray used for other variants.
- The `is-link` button, when disabled, is now gray instead of blue. I'm not too concerned about this change because, first of all, links can't be _really_ disabled. They should not have a disabled styling to start with. This is under discussion in https://github.com/WordPress/gutenberg/issues/62281. Anyways, I don't think it should stay blue (although dimmed) when 'disabled' because, historically, in WordPress the color blue indicates an _enabled_ interactive control.


## Testing Instructions





- Switch to this branch and run the Storybook: `npm run storybook:dev`
- The Storybook should automatically launch your browser and run at http://localhost:50240/
- To compare with the current styles, you can use the online version of the Storybook at 
https://wordpress.github.io/gutenberg/
- Test all the Button variants:
  - Default 
  - Primary
  - Secondary 
  - Tertiary 
  - Link 
  - Is Destructive 
  - Icon 
- First, test all the Button variants in their default state and observe there are no visual changes. Test the hover, focus, and active states.
- Then, test with only the `disabled` prop set to `true`.
    - Observe there are no visual changes exceot the ones mentioned above.
    - Test the hover and active states: observe the styling doesn't change when interacting with the button.
- Then, test with both the `disabled` and `__experimentalIsFocusable` props set to `true`.
    - Focus the buttons and observe the focus outline is always a solid color and has no opacity any longer.
    - Test the hover, focus, and active states: observe the styling doesn't change when interacting with the buttons, except for the focus outline.


### Testing Instructions for Keyboard


## Screenshots or screencast 

Example screenshots to illustrate:

Before:

![before](https://github.com/WordPress/gutenberg/assets/1682452/ca6ae535-bc5a-4816-b303-8cb128e7489b)


After: also arira-disabled buttons have a fully visible, consistent, focus outline:

![after](https://github.com/WordPress/gutenberg/assets/1682452/ebb6c2e4-b857-4a10-8979-ae93a3907497)
